pkg/util/usage/usage.go

```diff
@@ -33,6 +33,11 @@ func Usage(printAll bool, configs ...interface{}) error {
 		fieldCat := fieldcategory.Basic
 		var field reflect.StructField
 
+		// Do not print usage for deprecated flags.
+		if fl.Value.String() == "deprecated" {
+			return
+		}
+
```

Comment:
Which fields are affected by this change?

Reply:
It only affects `-distributor.extend-writes` and `-ingester.ring.join-after` (both deprecated). See diff here:
https://github.com/grafana/mimir/pull/1965/files#diff-4f928b957dd62bb4f119d1b1d69d3eb8965c672a5e9313a52cbee6b4fb58f33d